package NEgoProto;

import "model.proto";
import "strategy.proto";

enum EVariableType {
	FLOAT = 0;
	INT = 1;
	ENUM = 2;
}

message TVariable {
	optional string Name = 1;
	optional double Min = 2;
	optional double Max = 3;
	optional EVariableType Type = 4;
	repeated string Option = 5;
}

message TProblemConfig {
	optional string Name = 1;
	repeated TVariable Variable = 2;
}

message TProblemSpec {
	optional TProblemConfig ProblemConfig = 1;
	optional TModelConfig ModelConfig = 2;
	optional TStrategyConfig StrategyConfig = 3;
}

message TProblemState {
	optional TProblemConfig ProblemConfig = 1;
	optional TModelState ModelState = 2;
	optional TStrategyState StrategyState = 3;
}